src/agent/onefuzz-telemetry/src/lib.rs

```diff
@@ -6,10 +6,13 @@ use uuid::Uuid;
 
 pub use appinsights::telemetry::SeverityLevel::{Critical, Error, Information, Verbose, Warning};
 
+pub type MicrosoftTelemetryKey = Uuid;
```

Comment:
Using a `type` alias is good documentation, but still lets us use the wrong key in the wrong place.

To get typeck to help us, we'd need to define types like:
```rust
// We could make the inner field `pub`, or be safer and add accessors/`From` impls (at the cost of boilerplate).
pub struct MicrosoftTelemetryKey(Uuid);
pub struct InstanceTelemetryKey(Uuid);
```
And then ideally update the config definitions to deserialize each UUID directly as the correct type. This is made easy in a struct context via the serde [`#[serde(transparent)]`](https://serde.rs/container-attrs.html#transparent) attribute.

Reply:
added, please take a look